[quantization] __reduce__ for QScheme

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24969 [quantization] __reduce__ for QScheme**


This allows pickling qscheme objects

Differential Revision: [D16946567](https://our.internmc.facebook.com/intern/diff/D16946567)

Commit:
[quantization] __reduce__ for QScheme

raghuramank100 commented on test/test_quantized_tensor.py:
```diff
@@ -3,10 +3,19 @@
+class Foo(torch.nn.Module):
+    def __init__(self):
+        super(Foo, self).__init__()
+        self.dtype = torch.per_tensor_symmetric
```
per_tensor_symmetric should be a qscheme, not a dtype.  x.to(torch.per_tensor_symmetric) should be an error.
We should not be able to do x.to(torch.qint8), even though qint8 is a valid dtype as we do not have a scale and zero-point.

jamesr66a replied:
Oh this is just left over from me copy-pasting everything about dtype

raghuramank100 reviewed (dismissed):
Please see comments, thanks

Commit:
Update on "[quantization] __reduce__ for QScheme"



This allows pickling qscheme objects

Commit:
Update on "[quantization] __reduce__ for QScheme"



This allows pickling qscheme objects

Differential Revision: [D16946567](https://our.internmc.facebook.com/intern/diff/D16946567)

Commit:
Update on "[quantization] __reduce__ for QScheme"



This allows pickling qscheme objects

Differential Revision: [D16946567](https://our.internmc.facebook.com/intern/diff/D16946567)

Commit:
Update on "[quantization] __reduce__ for QScheme"



This allows pickling qscheme objects

Differential Revision: [D16946567](https://our.internmc.facebook.com/intern/diff/D16946567)
